Data/models/DataController.swift

```diff
@@ -235,49 +237,24 @@ public class DataController {
         return DataController.sharedInMemory.container.viewContext
     }
     
-    static func save(context: NSManagedObjectContext?) {
-        guard let context = context else {
-            log.warning("No context on save")
-            return
-        }
-        
-        if context.concurrencyType == .mainQueueConcurrencyType {
-            log.warning("Writing to view context, this should be avoided.")
-        }
-        
-        context.perform {
-            if !context.hasChanges { return }
-            
-            do {
-                try context.save()
-            } catch {
-                assertionFailure("Error saving DB: \(error)")
-            }
-        }
-    }
-    
-    func addPersistentStore(for container: NSPersistentContainer, store: URL) {
-        let storeDescription = NSPersistentStoreDescription(url: store)
-        
-        // This makes the database file encrypted until device is unlocked.
-        let completeProtection = FileProtectionType.complete as NSObject
-        storeDescription.setOption(completeProtection, forKey: NSPersistentStoreFileProtectionKey)
-        
-        container.persistentStoreDescriptions = [storeDescription]
-    }
-    
     var storeURL: URL {
         let supportDirectory = Preferences.Database.DocumentToSupportDirectoryMigration.completed.value
         return supportDirectory ? supportStoreURL : oldDocumentStoreURL
     }
     
     private func configureContainer(_ container: NSPersistentContainer, store: URL) {
-        addPersistentStore(for: container, store: store)
-        
-        // Dev note: This completion handler might be misleading: the persistent store is loaded synchronously by default.
-        container.loadPersistentStores(completionHandler: { _, error in
+        container.loadPersistentStores(completionHandler: { store, error in
             if let error = error {
-                fatalError("Load persistent store error: \(error)")
+                // Do not crash the app if the store already exists.
+                if (error as NSError).code != Self.storeExistsErrorCode {
+                    fatalError("Load persistent store error: \(error)")
```

Comment:
Can we log the error here isntead of giving a fatalError? Is there a benefit for us to crash application ?

Reply:
no, you should not allow the app to run if the database did not load correctly, 
only exception is the error code i added